model/common/tests/common/states/unit_tests/test_model_options.py

```diff
@@ -0,0 +1,90 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import functools
+import typing
+
+import pytest
+
+from icon4py.model.atmosphere.diffusion.diffusion_utils import scale_k
+from icon4py.model.common import model_backends
+from icon4py.model.common.model_options import customize_backend, setup_program
+
+
+@pytest.mark.parametrize(
+    "backend_factory, expected_backend",
+    [
+        (model_backends.make_custom_gtfn_backend, "gtfn"),
+        (model_backends.make_custom_dace_backend, "dace"),
+    ],
+)
+def test_custom_backend_options(backend_factory: typing.Callable, expected_backend: str) -> None:
+    backend_options: dict = {
+        "backend_factory": backend_factory,
+        "device": model_backends.CPU,
+    }
+    backend = customize_backend(backend_options)
+    backend_name = expected_backend + "_cpu"
+    assert str(model_backends.BACKENDS[backend_name]) == str(backend)
+
+
+def test_custom_backend_device() -> None:
```

Comment:
this test could probably be merged with the previous one

Reply:
I prefer to keep them separate, but I have adjusted to the new assert format + TODO